lib/tasks/config.rake

```diff
@@ -100,19 +100,19 @@ BANNER
       @keys.each do |key|
         value = @key_values[key]
         setting = Setting.find_by_name(key)
-        old_value = setting.value
         if value == :unset
           value = nil
         elsif complex_type?(setting.settings_type)
           setting.value = typecast_value(setting.settings_type, value)
         else
-          setting.parse_string_value(value)
+          if setting.parse_string_value(value)
```

Comment:
shouldn't there be something inside the if body?  If no, unless should be used intead